core/app/[locale]/not-found.tsx

```diff
@@ -41,11 +42,9 @@ export default async function NotFound() {
     <>
       <Header />
 
+      <NotFoundSection subtitle={t('message')} title={t('heading')} />
```

Comment:
Left it outside of `main` element for now, as my next PR will remove the `main` element and replace it with the `FeaturedProductsCarousel`